### PR TITLE
add cronjob for periodically getting all movie pages

### PR DESCRIPTION
**Pull request by menkuichan**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 const http = require('http');
 const axios = require('axios');
 const mongoose = require('mongoose');
+var CronJob = require('cron').CronJob;
 require('dotenv').config();
 
+new CronJob('0 0 12 * * * *', () => putAllMoviePagesInDatabase(), null, true, 'America/Los_Angeles');
```
Comment (ok2ju) [inline index.js]: You trying to start Cron and than save movies, but you not sure that db connection established

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const http = require('http');
 const axios = require('axios');
 const mongoose = require('mongoose');
+var CronJob = require('cron').CronJob;
```
Comment (ok2ju) [inline index.js]: Why do you use `var` here? Please use modern approach, e.g. `const { CronJob } = require('cron');`

File: index.js [hunk 3]
```diff
@@ -69,6 +62,19 @@ db.once('open', async () => {
...
+mongoose.connect(process.env.DB_URI, { useNewUrlParser: true, useUnifiedTopology: true });
+
+const db = mongoose.connection;
+
+db.on('error', (err) => {
+  console.error(err);
+});
+
+db.once('open', () => {
+  new CronJob('0 0 12 * * * *', () => putAllMoviePagesInDatabase(), null, true, 'America/Los_Angeles');
```
Comment (ok2ju) [inline index.js]: Why do you use `America/Los_Angeles` locale? Maybe try to use our local locale?

File: index.js [hunk 4]
```diff
@@ -69,6 +62,19 @@ db.once('open', async () => {
...
+  putAllMoviePagesInDatabase();
```
Comment (ok2ju) [inline index.js]: You don't need to call `putAllMoviePagesInDatabase` here. Only inside cron callback.

File: index.js [hunk 5]
```diff
@@ -45,15 +46,7 @@ const movieScheme = new Schema({
...
-mongoose.connect(process.env.DB_URI, { useNewUrlParser: true, useUnifiedTopology: true });
-
-const db = mongoose.connection;
-
-db.on('error', (err) => {
-  console.error(err);
-});
-
-db.once('open', async () => {
+const putAllMoviePagesInDatabase = async () => {
```
Comment (ok2ju) [inline index.js]: Let's name this function something like `populateDB`. WDYT?

File: index.js (hunk 4 above)
Comment (ok2ju) [reply index.js]: If you want to start cron immediately use `start` method of `cron` - [example](https://github.com/kelektiv/node-cron/blob/master/examples/every_10_minutes.js)

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: index.js [hunk 6]
```diff
@@ -69,6 +62,19 @@ db.once('open', async () => {
...
+db.on('error', (err) => {
+  console.error(err);
+});
+
+db.once('open', () => {
+  new CronJob('0 0 12 * * * *', () => putAllMoviePagesInDatabase(), null, true, 'America/Los_Angeles');
+  putAllMoviePagesInDatabase();
 });
 
 const server = http.createServer((req, res) => {
```
Comment (ok2ju) [inline index.js]: I think we can remove this redundant callback now. Just start server without callback.

Comment (ok2ju) [inline index.js]: I think we can remove this redundant callback now. Just start server without callback.

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: index.js [hunk 7]
```diff
@@ -46,7 +46,7 @@ const movieScheme = new Schema({
 
 const Movie = mongoose.model('Movie', movieScheme);
 
-const putAllMoviePagesInDatabase = async () => {
+const populateDB = async () => {
   try {
     let allMovies = [];
     const { results, total_pages } = await getMovies(1);
```
Comment (menkuichan) [inline index.js]: @ok2ju is it correct use 'results' inside for? because 'results' is already declared in the upper scope.

Comment (menkuichan) [reply index.js]: should i just rename it?

**Review by ok2ju — APPROVED** [review]

**Pull request merged**